scripts/ai/AIDriveStrategyUnloadCombine.lua

```diff
@@ -619,24 +652,37 @@ function AIDriveStrategyUnloadCombine:getCourseToAlignTo(vehicle, offset)
     return tempCourse
 end
 
-function AIDriveStrategyUnloadCombine:getTrailersTargetNode()
-    local trailer = AIUtil.getImplementOrVehicleWithSpecialization(self.vehicle, Trailer)
-    if trailer then
-        if self.combineToUnload:getCpDriveStrategy():canLoadTrailer(trailer) then
-            local targetNode = trailer:getFillUnitAutoAimTargetNode(1)
-            if targetNode then
-                return targetNode
-            else
-                self:debugSparse('Can\'t find trailer target node')
+--- Gets fill nodes of all fill units.
+---@return table|nil
+function AIDriveStrategyUnloadCombine:getTrailersTargetNodes()
+    local targetNodes = {}
+    local trailers = AIUtil.getAllChildVehiclesWithSpecialization(self.vehicle, Trailer)
+    if not trailers then 
+        self:debugSparse('Can\'t find any trailers')
+        return
+    end
+    for _, trailer in pairs(trailers) do 
+        for ix, _ in pairs(trailer:getFillUnits()) do 
+            local node = trailer:getFillUnitExactFillRootNode(ix)
+            local _, _, trailerOffset = localToLocal(node, trailer.rootNode, 0, 0, 0)
+            local _, _, vehicleOffset = localToLocal(trailer.rootNode, self.vehicle.rootNode, 0, 0, 0)
```

Comment:
If there is no guarantee that the tractor is not aligned with the trailer then I'd probably flip the nodes here to get the offset in the trailer's coordinate system for the case when the angle between the tractor and trailer is over 90 degrees.